indra/llcharacter/llik.cpp

```diff
@@ -577,6 +651,121 @@ size_t LLIK::ElbowConstraint::generateHash() const
     return seed;
 }
 
```

Comment:
The `ElbowConstraint::enforce()` override is a big contributor to higher stability and accuracy of the arm IK algorithm.  Basically what it does is: rather than change the direction of the lower-arm in order to enforce the constraint at the elbow it assumes its direction is correct and instead spins the upper- and lower-arm (shoulder, elbow) joints to line up the elbow's pivot axis in each's local-frame, all within the twist limits of the elbow's forearm.  Besides providing faster convergence of the IK iterations this also provides effective "back-pressure" on the shoulder joint when the lower arm can't quite reach its orientation target, which would allow lower joints to move in an attempt to help reach the goal... although we are not yet taking full advantage of that since the shoulder doesn't have any twist constraints ATM so most back-pressure doesn't propagate lower than the shoulder.